   Copyright (C) 2010 by Frederic-Charles Barthelery,
                         Jean-Manuel Da Silva,
                         Nikita Kozlov,
                         Philippe Lago,
                         Jean Baptiste Vergely,
                         Vincent Veronis.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

--- ../../../src/smack/org/jivesoftware/smack/ChatManager.java	2010-02-23 19:27:26.000000000 +0100
+++ org/jivesoftware/smack/ChatManager.java	2010-02-23 19:37:47.000000000 +0100
@@ -111,7 +111,9 @@
                     	chat = getUserChat(message.getFrom());
                     }
                 }
-
+		if (chat == null) {
+                	chat = getUserChat(StringUtils.parseBareAddress(message.getFrom()));
+		}
                 if(chat == null) {
                     chat = createChat(message);
                 }
